### PR TITLE
refactor(store-engine): Map Dexie errors

### PR DESCRIPTION
**Pull request by bennyn**
## Pull Request Checklist

- [x] My code is covered by tests
- [x] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/store-engine/src/main/engine/IndexedDBEngine.ts [hunk 1]
```diff
@@ -78,20 +78,26 @@ export default class IndexedDBEngine implements CRUDEngine {
     return this.db ? this.db.delete() : Dexie.delete(this.storeName);
   }
 
+  private mapDatabaseError(error: Dexie.DexieError, tableName: string, primaryKey: string): Error {
+    const isAlreadyExisting = error instanceof Dexie.ConstraintError;
+    const hasNotEnoughDiskSpace =
```
[the remark this answers is not in the document]
Comment (bennyn) [reply packages/store-engine/src/main/engine/IndexedDBEngine.ts]: I wanted to stick to the convention to prefix booleans with `is`, `has` and similar: https://softwareengineering.stackexchange.com/a/232217/58311

[the remark this answers is not in the document]
Comment (bennyn) [reply packages/store-engine/src/main/engine/IndexedDBEngine.ts]: We can make it `hasLowDiskSpace`.

**Comment by ffflorian** [comment]
Please do not merge before https://github.com/wireapp/wire-web-packages/pull/1277 or merge it as breaking change.

File: packages/store-engine/src/main/engine/IndexedDBEngine.ts [hunk 2]
```diff
@@ -78,20 +78,27 @@ export default class IndexedDBEngine implements CRUDEngine {
     return this.db ? this.db.delete() : Dexie.delete(this.storeName);
   }
 
+  private mapDatabaseError(error: Dexie.DexieError, tableName: string, primaryKey: string): Error {
+    const isAlreadyExisting = error instanceof Dexie.ConstraintError;
+    // @see https://github.com/dfahlander/Dexie.js/issues/776
```
Comment (ffflorian) [inline packages/store-engine/src/main/engine/IndexedDBEngine.ts]:
```suggestion
    /** @see https://github.com/dfahlander/Dexie.js/issues/776 */
```

or
```suggestion
    // see https://github.com/dfahlander/Dexie.js/issues/776
```

**Pull request merged**